models/articles.js

```diff
@@ -26,3 +26,43 @@ exports.selectArticles = (sortBy = "created_at") => {
       throw err;
     });
 };
+
+exports.selectArticleById = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .then((result) => {
+      return result.rows[0];
+    });
+};
+
+exports.selectCommentsById = (article_id) => {
+  return db
+    .query(
+      `SELECT comment_id, votes, created_at, author, body, article_id 
+           FROM comments 
+           WHERE article_id = $1 
+           ORDER BY created_at DESC`,
+      [article_id]
+    )
+    .then((result) => {
+      return result.rows;
+    })
+    .catch((err) => {
+      console.error("Error in selectCommentsById:", err);
+      throw err;
+    });
+};
+
+exports.checkArticleExists = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        const err = new Error("Article not found");
+        err.status = 404;
+        err.msg = "Article not found";
+        throw err;
+      }
+      return true;
+    });
+};
```

Comment:
This model is almost identical to selectArticleById, except with some error handling which to me begs the question of why not add this error handling to selectArticleById and then reuse that function. selectArticleById would still want to be able to send a 404 back if the article wasn't found, so this approach would makes sense :)

Reply:
thanks Modou I'll have a refactor with the DRY principle in mins